perceval/components/unitary_components.py

```diff
@@ -453,30 +441,34 @@ def __init__(self):
                     [0, 0, 0, 1]])
         super().__init__(U=u, use_polarization=True)
 
-    def get_variables(self, map_param_kid=None):
-        return {}
-
     # noinspection PyMethodMayBeStatic
-    def describe(self, _=None):
+    def describe(self):
         return "PBS()"
 
 
-class Barrier(Unitary):
+class Barrier(ACircuit):
     """Behaves like an identity unitary, visually represented as a barrier."""
     DEFAULT_NAME = "I"
 
-    def __init__(self, num_modes):
-        assert isinstance(num_modes, int), "identity Operator needs list parameter"
-        self.num_modes = num_modes
-        u = Matrix.eye(num_modes, use_symbolic=False)
-        super().__init__(U=u)
+    def __init__(self, num_modes: int, visible: bool = True):
+        assert isinstance(num_modes, int), "Barrier"
+        self._visible = bool(visible)
+        super().__init__(num_modes)
 
-    def describe(self, _=None):
-        return f"Barrier({self.num_modes})"
+    @property
+    def visible(self):
```

Comment:
Maybe add a setter also to allow the user to have invisible barriers ?

Reply:
`visible` is already a parameter of `Barrier.__init__`

Reply:
Yes but when you create a barrier with the operator @ you don't have the possibility to make it invisible

Reply:
You don't have to use @ operator. It's only a facilitator for a very very specific case